dotcom-rendering/src/server/render.front.web.tsx

```diff
@@ -17,9 +18,61 @@ interface Props {
 	front: DCRFrontType;
 }
 
+const extractFrontNav = (front: DCRFrontType): NavType => {
+	const NAV = extractNAV(front.nav);
+	const { currentNavLink } = NAV;
+
+	// Is the `currentNavLink` a pillar?
+	const pillarFromCurrentLink = (() => {
+		switch (currentNavLink) {
+			// The pillar name is "arts" in CAPI, but "culture" everywhere else
+			case 'Arts':
+			case 'Culture':
+				return ArticlePillar.Culture;
+			case 'Opinion':
+				return ArticlePillar.Opinion;
+			case 'News':
+				return ArticlePillar.News;
+			case 'Sport':
+				return ArticlePillar.Sport;
+			case 'Lifestyle':
+				return ArticlePillar.Lifestyle;
+			default:
+				return undefined;
+		}
+	})();
+
+	// Is the `currentNavLink` in one of the children of the pillar?
+	const themeFromSubNav = NAV.pillars.find((pillar) => {
+		// Annoyingly "Football" appears in "News" and "Sport" pillars, so we exclude this case in "News"
+		// As "Football" is always "Sport". You can see the corresponding `frontend` code here:
+		// https://github.com/guardian/frontend/blob/main/common/app/navigation/Navigation.scala#L141-L143
+		if (
+			pillar.pillar === ArticlePillar.News &&
+			currentNavLink === 'Football'
+		) {
+			return false;
+		}
+
+		return pillar.children?.some((child) => {
+			return child.title === currentNavLink;
+		});
+	})?.pillar;
```

Comment:
Not a blocker, could be a follow-up, but it might be useful to capture this logic in a standalone function so we can write some tests for it?